packages/beacon-node/src/api/impl/beacon/state/index.ts

```diff
@@ -130,6 +130,35 @@ export function getBeaconStateApi({
       return this.getStateValidators(args, context);
     },
 
+    async postStateValidatorIdentities({stateId, validatorIds = []}) {
+      const {state, executionOptimistic, finalized} = await getStateResponse(chain, stateId);
+      const {pubkey2index} = chain.getHeadState().epochCtx;
```

Comment:
should we get `pubkey2index` from `state` instead?

Reply:
do you mean to use something like `chain.getHeadStateAtEpoch()`? I think for most cases uses the pubkey<>index map from head state should be fine.

Do you have concerns if this api is called for historic state where the validator might be exited and not part of index map anymore from head state?

Reply:
> Do you have concerns if this api is called for historic state where the validator might be exited and not part of index map anymore from head state?

In the current design, I think it's safe to assume `EpochCache.pubkey2index` is append-only. So no modification and deletion to existed cache entry.

I think the current approach ie. the combination of `getStateValidatorIndex()` and `state.validators.getReadonly(i)` is fine for current use.

Reply:
I mean `const {pubkey2index} = state.epochCtx;`, but they are the same to head state anyway

Reply:
the `state` here is an ssz view and does not have `epochCtx`